Tasks/Kubernetes/Tests/TestSetup.ts

```diff
@@ -0,0 +1,235 @@
+    "code": 0,
+    "stdout": "successfully listed the environment variables defined on all pods"
+},
+a.exec[`kubectl --kubeconfig ${KubconfigFile} top node`] = {
+    "code": 0,
+    "stdout": "successfully showed metrics for all nodes"
+},
+a.exec[`kubectl --kubeconfig ${KubconfigFile} delete secret my-secret`] = {
+    "code": 0
+};
+a.exec[`kubectl --kubeconfig ${KubconfigFile} create secret docker-registry my-secret --docker-server=ajgtestacr1.azurecr.io --docker-username=spId --docker-password=spKey --docker-email=ServicePrincipal@AzureRM`] = {
+    "code": 0
+};
+a.exec[`kubectl --kubeconfig ${KubconfigFile} create secret docker-registry my-secret --docker-server=https://index.docker.io/v1/ --docker-username=test --docker-password=regpassword --docker-email=test@microsoft.com`] = {
+    "code": 0
+};
+a.exec[`kubectl --kubeconfig ${KubconfigFile} get secrets my-secret -o yaml`] = {
+    "code": 0,
+    "stdout": "successfully got secret my-secret and printed it in the specified format"
+};
+
+tr.setAnswers(<any>a);
+
+// Create mock for fs module
+let fs = require('fs');
+let fsClone = Object.assign({}, fs);
+fsClone.existsSync = function(filePath) {
+    switch (filePath) {
+        case "kubectl":
+            if(JSON.parse(process.env[shared.isKubectlPresentOnMachine]))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            } 
+        case KubectlPath:
+            return true;     
+        case KubconfigFile:
+            return true;
+        case ConfigurationFilePath:
+            return true;
+        default:
+            return fs.existsSync(filePath);
+    }
+};
+fsClone.writeFileSync = function(fileName, data) {
+    switch (fileName) {
+        case KubconfigFile:
+            console.log("Content of kubeconfig file : " + data); 
+            break;
+        default:
+            return fs.writeFileSync(fileName, data);
+    }
+};
+
+fsClone.chmod = function(path, mode) {
+      switch(path){
+          case KubectlPath:
+            console.log("Added permissions to kubectlPath");
+            break;
+          default:
+            fs.chmod(path, mode);        
+      }
+};
+tr.registerMock('fs', fsClone);
+
+var ut = require('../src/utilities');
+tr.registerMock('./utilities', {
+    IsNullOrEmpty : ut.IsNullOrEmpty,
+    HasItems : ut.HasItems,
+    StringWritable: ut.StringWritable,
+    PackerVersion: ut.PackerVersion,
+    isGreaterVersion: ut.isGreaterVersion,
+    getTempDirectory: function() {
+        return newUserDirPath;
+    },
+    getCurrentTime: function() {
+        return new Date(1996,3,5);
+    },
+    getNewUserDirPath: function() {
+        return newUserDirPath;
+    },
+    getStableKubectlVersion: function(){
+        return "v1.6.6";
+    },
+    getKubectlVersion: async function(versionSpec, checkLatest) : Promise<string> {
+        let version: string = "v1.6.6";   
```

Comment:
We should avoid having logic in mocq.

Reply:
Need to add the logic as the mock getStableKubectlVersion() will not be called if we use the original getKubectlVersion().